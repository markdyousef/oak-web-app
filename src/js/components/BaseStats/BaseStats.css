:root {
    --main-black-color: #000;
    --main-white-color: #fefeff;
    --main-orange-color: #ff9200;
    --main-red-color: #8b0000;
    --main-green-color: #005d6c;
}

.container {}

.statsGrid {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
}
