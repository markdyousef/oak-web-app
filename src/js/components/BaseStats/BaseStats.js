// @flow
import React, { Component, PropTypes } from 'react';

import ChannelStats from '../ChannelStats';
import Toolbar from '../../containers/ToolbarContainer';

import css from './BaseStats.css';

class BaseStats extends Component {
    static propTypes = {
        getTeam: PropTypes.func.isRequired,
        channels: PropTypes.arrayOf(
            PropTypes.shape({
                name: PropTypes.string.isRequired
            })
        ).isRequired
    }
    constructor() {
        super();
        this.renderStats = this.renderStats.bind(this);
    }
    componentWillMount() {
        const { getTeam } = this.props;
        getTeam('clai');
    }
    renderStats() {
        const { channels } = this.props;
        // group data by channel
        const groupedChannels = [...channels.map(channel =>
            ({ name: channel.name, data: channel }))
        ]
        return groupedChannels.map(channel =>
            <ChannelStats
                key={channel.name}
                data={channel.data}
                name={channel.name}
                id={channel.id}
            />
            );
    }
    render() {
        return (
            <div className={css.container}>
                <div className={css.toolbar}>
                    <Toolbar />
                </div>
                <div className={css.statsGrid}>
                    {this.renderStats()}
                </div>
            </div>
        );
    }
}

export default BaseStats;
