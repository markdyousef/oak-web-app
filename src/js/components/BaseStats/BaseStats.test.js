// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import BaseStats from './BaseStats';

test('renders BaseStats', () => {
    const props = {
        data: []
    };
    const component = renderer.create(
        <BaseStats {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
