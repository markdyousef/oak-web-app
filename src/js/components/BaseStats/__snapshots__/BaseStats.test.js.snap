exports[`test renders BaseStats 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <div
        className={undefined}>
        <Connect(ToolBar) />
    </div>
    <div
        className={undefined}>
        <div>
            No Channels
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <div
            className={undefined}>
            <Connect(ToolBar) />
      </div>
      <div
            className={undefined}>
            <div>
                  No Channels
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <BaseStats
        channels={Array []}
        getTeam={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": BaseStats {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "channels": Array [],
          "getTeam": [Function],
        },
        "refs": Object {},
        "renderStats": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <div
                    className={undefined}>
                    <Connect(ToolBar) />
          </div>
          <div
                    className={undefined}>
                    <div>
                              No Channels
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <div
                    className={undefined}>
                    <Connect(ToolBar) />
          </div>
          <div
                    className={undefined}>
                    <div>
                              No Channels
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BaseStats
    channels={Array []}
    getTeam={[Function]} />,
}
`;
